src/main/java/org/apache/flink/benchmark/UnalignedCheckpointTimeBenchmark.java

```diff
@@ -42,119 +43,90 @@
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.flink.api.common.eventtime.WatermarkStrategy.noWatermarks;
 
 /**
- * The benchmark for measuring the time taken to finish the configured number of
- * unaligned checkpoints.
+ * The benchmark for measuring the time taken to finish the configured number of unaligned
+ * checkpoints.
  */
-@OutputTimeUnit(MINUTES)
-@OperationsPerInvocation(value = UnalignedCheckpointTimeBenchmark.NUM_FINISHED_CHECKPOINTS)
+@OutputTimeUnit(SECONDS)
+@OperationsPerInvocation(UnalignedCheckpointTimeBenchmark.NUM_FINISHED_CHECKPOINTS)
 public class UnalignedCheckpointTimeBenchmark extends BenchmarkBase {
-    public static final int NUM_FINISHED_CHECKPOINTS = 5;
+    public static final int NUM_FINISHED_CHECKPOINTS = 10;
     private static final int NUM_VERTICES = 3;
     private static final int PARALLELISM = 4;
     private static final long CHECKPOINT_INTERVAL_MS = 10;
 
     public static void main(String[] args) throws RunnerException {
-        Options options = new OptionsBuilder()
-            .verbosity(VerboseMode.NORMAL)
-            .include(UnalignedCheckpointTimeBenchmark.class.getCanonicalName())
-            .build();
+        Options options =
+                new OptionsBuilder()
+                        .verbosity(VerboseMode.NORMAL)
+                        .include(UnalignedCheckpointTimeBenchmark.class.getCanonicalName())
+                        .build();
 
         new Runner(options).run();
     }
 
     @Benchmark
-    public void unalignedCheckpoint(UnalignedCheckpointEnvironmentContext context) throws Exception {
+    public void unalignedCheckpoint(UnalignedCheckpointEnvironmentContext context)
+            throws Exception {
         StreamExecutionEnvironment env = context.env;
-        DataStreamSource<byte[]> source = env.addSource(new FiniteCheckpointSource(NUM_FINISHED_CHECKPOINTS));
-        source
-            .slotSharingGroup("source").rebalance()
-            .map((MapFunction<byte[], byte[]>) value -> value).slotSharingGroup("map").rebalance()
-            .addSink(new SlowDiscardSink<>()).slotSharingGroup("sink");
+        DataStreamSource<Record> source =
+                env.fromSource(
+                        new RecordSource(NUM_FINISHED_CHECKPOINTS),
+                        noWatermarks(),
+                        RecordSource.class.getName());
+
+        source.slotSharingGroup("source")
+                .rebalance()
+                .map((MapFunction<Record, Record>) value -> value)
+                .slotSharingGroup("map")
+                .rebalance()
+                .addSink(new SlowDiscardSink<>())
+                .slotSharingGroup("sink");
 
         env.execute();
     }
 
     public static class UnalignedCheckpointEnvironmentContext extends FlinkEnvironmentContext {
 
-        @Param({"REMOTE", "LOCAL"})
-        public String mode = "REMOTE";
+        @Param({"0", "1", "ALIGNED"})
```

Comment:
I think so. `ALIGNED` and `0` do not test timeouts whatsoever, so we need some setting that would benchmark the timeouts. The lowest possible timeout setting (`1ms`) makes the most sense, as we would ideally be expecting the results be almost as good (+/- 1ms) as fully unaligned checkpoints. Any setting higher than that, and we are actually benchmarking sleep times, instead of efficiency/overhead of the time outing algorithm.